api/server.go

```diff
@@ -22,23 +22,47 @@ type API struct {
 	agents map[string]*loadtest.LoadTester
 }
 
-func writeJsonResponse(w http.ResponseWriter, status int, data interface{}) {
+// APIResponse contains the data returned by the HTTP server.
+type APIResponse struct {
+	Message string          `json:"message,omitempty"` // Message contains information about the response.
+	Status  loadtest.Status `json:"status"`            // Status contains the current status of the load test.
+	Error   error           `json:"error,omitempty"`   // Error is set if there was an error during the operation.
+}
+
+func writeResponse(w http.ResponseWriter, status int, response *APIResponse) {
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(status)
-	_ = json.NewEncoder(w).Encode(data)
+	_ = json.NewEncoder(w).Encode(response)
 }
 
 func (a *API) createLoadAgentHandler(w http.ResponseWriter, r *http.Request) {
 	var config config.LoadTestConfig
 	err := json.NewDecoder(r.Body).Decode(&config)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+		writeResponse(w, http.StatusBadRequest, &APIResponse{
+			Error: err,
+		})
+		return
+	}
+
+	if config.ConnectionConfiguration.ServerURL == "" {
+		writeResponse(w, http.StatusBadRequest, &APIResponse{
+			Error: fmt.Errorf("ServerURL is not present in config"),
+		})
+		return
+	}
+	if config.ConnectionConfiguration.WebSocketURL == "" {
+		writeResponse(w, http.StatusBadRequest, &APIResponse{
+			Error: fmt.Errorf("WebSocketURL is not present in config"),
+		})
```

Comment:
Maybe it makes more sense to have a `LoadTestConfig.IsValid()` method in `config.go` and have that return an error if something is not valid.